File: backend/API/Scene.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;


namespace API


{
    public class Scene
    {

        public Scene()
        {
        }
        [Key]
        public int Id { get; set; }

        public string Title { get; set; }
        public string Description { get; set; }
        public int Order { get; set; }

        [ForeignKey("Chapter")]
        public int ChapterId { get; set; }
        public Chapter Chapter { get; set; }
    }
}

File: src/app/scene.ts
import { Statement } from "@angular/compiler";


export interface Scene {
    id?: number;
    title: string;
    description: string;
    order: number;
    chapterId: number;
  }
  
  export interface Narrative extends Scene {
    events: string[];
  }

  export interface Dialogue extends Scene {
    statements: Statement[];
  }